test/e2e_node/system_node_critical_test.go

```diff
@@ -110,6 +110,13 @@ var _ = framework.KubeDescribe("SystemNodeCriticalPod [Slow] [Serial] [Disruptiv
 					return checkMirrorPodDisappear(f.ClientSet, mirrorPodName, ns)
 				}, time.Minute, time.Second*2).Should(gomega.BeNil())
 
+				ginkgo.By("making sure that node no longer has DiskPressure")
```

Comment:
Can you share more about why the `defer` statement on line 97 doesn't have the same checks that we added to the other `defer` statement?

Reply:
Not sure if I understand your question. :thinking: 

If you mean the `expectedNodeCondition == v1.NodeDiskPressure` check, that is only relevant to `eviction_tests` and not `system_node_critical_tests` afaik, or?

If you think we should add the diskPressure "wait" to the defer, I think we need to add some custom error handling to make sure we always run PrePull, even tho. node still have `NodeDiskPressure`.

Reply:
Ah yes, that was my questions :)